pkg/block/s3/adapter.go

```diff
@@ -117,20 +97,106 @@ func WithServerSideEncryptionKmsKeyID(s string) func(a *Adapter) {
 
 type AdapterOption func(a *Adapter)
 
-func NewAdapter(awsSession *session.Session, opts ...AdapterOption) *Adapter {
+func NewAdapterFromParams(ctx context.Context, params params.S3, opts ...AdapterOption) (*Adapter, error) {
+	cfg, err := LoadConfig(ctx, params)
+	if err != nil {
+		return nil, err
+	}
+	a := &Adapter{
+		clients:         NewClientCache(cfg, params),
+		preSignedExpiry: block.DefaultPreSignExpiryDuration,
+	}
+	for _, opt := range opts {
+		opt(a)
+	}
+	return a, nil
+}
+
+func NewAdapter(awsConfig aws.Config, params params.S3, opts ...AdapterOption) *Adapter {
```

Comment:
Is this one still needed?

Reply:
No - thanks! remove the kept one way to create new block adapter.